.github/workflows/lint-whitespace.yml

```
@@ -0,0 +1,19 @@
+name: Lint whitespace
+
+on: [push, pull_request, workflow_dispatch]
```

You could consider doing:

```suggestion
on:
  pull_request:
  push:
    branches:
      - main
  workflow_dispatch:
```

That way, pushes to PRs are not doubly run.

---

The rationale for the current way is:
- I like to run CI by pushing to my fork (without needing to trigger via `workflow_dispatch`) before creating a PR. If I forgot to run `shellcheck` or `markdownlint` on something, it's nice to get a notification for that before creating the PR. So it can reduce initial noise in a PR, making things easier for a reviewer, and reduce the number of jobs that Exercism runs (limit of 20 concurrent jobs, I think). 
- GitHub claims a runtime of a few seconds for these workflows, so the damage to the planet is a small fraction of the other jobs in this repo.
- It's consistent with:

https://github.com/exercism/configlet/blob/4ab7a531a5968339e69d10990f0188e3c57d2e97/.github/workflows/markdownlint.yml#L3

https://github.com/exercism/configlet/blob/4ab7a531a5968339e69d10990f0188e3c57d2e97/.github/workflows/shellcheck.yml#L3

I've had some conversations about this in the past. I think the condition I want is "run on push, unless there's an open PR for that branch" - but at the time, it wasn't obvious to me how to do that. Have you ever done this? I'd be happy to change the workflows in this repo as such.

---

I've never done that. It might be possible by inspecting the `github.event` object. I'm okay with keeping it as is.

---

Tracking in #543.